packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -12,325 +12,271 @@ import 'package:flutter/services.dart';
-      case TargetPlatform.android:
-      case TargetPlatform.fuchsia:
-      case TargetPlatform.linux:
-      case TargetPlatform.windows:
-        _announceToAccessibility(context, MaterialLocalizations.of(context).postMeridiemAbbreviation);
-        break;
-      case TargetPlatform.iOS:
-      case TargetPlatform.macOS:
-        break;
-    }
-    _togglePeriod();
+  void _handleChangeMode(_TimePickerMode value) {
+    if (value != mode)
+      onModeChanged(value);
   }
 
   @override
   Widget build(BuildContext context) {
-    final MaterialLocalizations materialLocalizations = MaterialLocalizations.of(context);
-    final TextTheme headerTextTheme = fragmentContext.headerTextTheme;
-    final TimeOfDay selectedTime = fragmentContext.selectedTime;
-    final Color activeColor = fragmentContext.activeColor;
-    final Color inactiveColor = fragmentContext.inactiveColor;
-    final bool amSelected = selectedTime.period == DayPeriod.am;
-    final TextStyle amStyle = headerTextTheme.subtitle1.copyWith(
-      color: amSelected ? activeColor: inactiveColor
-    );
-    final TextStyle pmStyle = headerTextTheme.subtitle1.copyWith(
-      color: !amSelected ? activeColor: inactiveColor
+    assert(debugCheckHasMediaQuery(context));
+    final ThemeData themeData = Theme.of(context);
+    final TimeOfDayFormat timeOfDayFormat = MaterialLocalizations.of(context)
+        .timeOfDayFormat(alwaysUse24HourFormat: MediaQuery.of(context).alwaysUse24HourFormat);
+
+    final _TimePickerFragmentContext fragmentContext = _TimePickerFragmentContext(
+      selectedTime: selectedTime,
+      mode: mode,
+      onTimeChange: onChanged,
+      onModeChange: _handleChangeMode,
+      use24HourDials: use24HourDials,
     );
-    final bool layoutPortrait = orientation == Orientation.portrait;
 
-    final double buttonTextScaleFactor = math.min(MediaQuery.of(context).textScaleFactor, 2.0);
+    EdgeInsets padding;
+    double width;
+    Widget controls;
 
-    final Widget amButton = ConstrainedBox(
-      constraints: _kMinTappableRegion,
-      child: Material(
-        type: MaterialType.transparency,
-        child: InkWell(
-          onTap: Feedback.wrapForTap(() => _setAm(context), context),
-          child: Padding(
-            padding: layoutPortrait ? const EdgeInsets.only(bottom: 2.0) : const EdgeInsets.only(right: 4.0),
-            child: Align(
-              alignment: layoutPortrait ? Alignment.bottomCenter : Alignment.centerRight,
-              widthFactor: 1,
-              heightFactor: 1,
-              child: Semantics(
-                selected: amSelected,
-                child: Text(
-                    materialLocalizations.anteMeridiemAbbreviation,
-                    style: amStyle,
-                    textScaleFactor: buttonTextScaleFactor,
-                ),
+    assert(orientation != null);
+    switch (orientation) {
+      case Orientation.portrait:
+        padding = const EdgeInsets.symmetric(horizontal: 24.0);
```

Comment:
Maybe a comment here that width stays null for X reason?

Reply:
done